src/query/expression/src/kernels/take.rs

```diff
@@ -144,6 +173,244 @@ impl Column {
         }
     }
 
+    pub fn probe_take(&self, indices: &[(u32, u32)], indices_len: usize, probe_num: usize) -> Self {
+        let length = indices.len();
+        match self {
+            Column::Null { .. } | Column::EmptyArray { .. } | Column::EmptyMap { .. } => {
+                self.slice(0..length)
+            }
+            Column::Number(column) => {
+                with_number_mapped_type!(|NUM_TYPE| match column {
+                    NumberColumn::NUM_TYPE(values) => Self::probe_take_arg_types::<
```

Comment:
numeric path could work with path `copy_nonoverlapping` like decimal

Reply:
numeric path will call probe_take_arg_types and will have the same behavior as decimal.